src/app/base/components/dropdown-iterative/dropdown-iterative.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { AbstractControl, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'kern-dropdown-iterative',
+  templateUrl: './dropdown-iterative.component.html',
+  styleUrls: ['./dropdown-iterative.component.scss']
+})
+export class DropdownIterativeComponent implements OnInit {
+
+  @Input() arrayOptions = [];
+  @Input() condition;
+  @Input() valueIfConditionTrue;
+  @Input() valueIfConditionFalse;
+  @Input() buttonTooltip: string;
+  @Input() labelingTasks;
+  @Input() property: string;
+  @Input() optionProperty: string;
+  @Input() disabledCondition: boolean;
+  @Input() hasCheckboxes: boolean;
+  @Input() formArrayName: string;
```

Comment:
i think we can reduce the amount of input parameter by using a type with optional parameters that hold the specification. I think something like " @Input() dropdownValues = []; @Input() dropdownOptions;" options in turn have values for "hasCheckbox?:boolean, useFormArray?:FormArray, svgArrag?:string[]...) so the basic version would only use the list of arrays but additional logic can be added via the array.

If it helps I could build a lite version to better show what I mean